### PR TITLE
Refactor APCs interaction chain from attackby to item_interaction

### PR DESCRIPTION
**Pull request by 00-Steven**
## About The Pull Request

For how many lines this is, there's not a lot to really say.
In general, we simply move all item interactions from `attackby(...)` to `item_interaction(...)`, split each item interaction off into a separate proc, and make them all return the proper item interaction flags.
We _do_ kill some probably dead code, and remove a call to `attackby(...)` elsewhere. Then, for clarity, we move the cell check below the ID check so it can be next to the other item type checks, as the priority between cell and ID is unlikely to matter anyway.
Other than what's described above and detailed below, each section's functionality should be the same.

Now, for the parts that _do_ need to be explained more.

### Killing Probably Dead Code

Alright, so, the first part that does not have the cleanest transition.
https://github.com/tgstation/tgstation/blob/d38f9385b863e49f83455a227764d302629e2867/code/modules/power/apc/apc_attack.dm#L22-L23
Whatever the fuck this is.

Asking around, this seems to just be dead code.
For sanity's sake removing it and testing, silicon interactions with it seem to work just fine.
So we kill it. We just kill it. We Just Kill It.

Closest we could find requires the distance check there to be false, so it wouldn't apply. But it _does_ bring us to the second bit of weird code.

### Calling APC Attackby Elsewhere?
So wallframes let you screwdriver them to put them up, which from a comment seems to be because of cyborgs.
APC wallframes of course override this with their own implementation, that allows you to also replace a damaged cover or frame like that!
https://github.com/tgstation/tgstation/blob/d38f9385b863e49f83455a227764d302629e2867/code/game/objects/items/apc_frame.dm#L29-L39
...By just calling the wholeass `attackby(...)` proc on the APC and calling it a day.

But hey, this is where our previous splitting up comes in handy, because we just have a `wallframe_act(...)` proc!
So we just call that instead.
```dm
var/obj/machinery/power/apc/mounted_apc = locate(/obj/machinery/power/apc) in get_turf(user)
mounted_apc.wallframe_act(user, src)
return ITEM_INTERACT_SUCCESS
```
...And not use single letter variables, while we're at it.

That should be all.
Remember to get snacks and drinks.
## Why It's Good For The Game

Split off 178 line `attackby(...)` item interaction chain into separate procs called in `item_interaction(...)`.
Screwdrivering APC wallframes no longer calls the wholeass `attackby(...)` on the APC, but just call the new sub-proc for the specific interaction it cares about.
## Changelog
:cl:
refactor: APCs have had their item interaction chain refactored. This should functionally be the same, but please report any issues.
/:cl:

### CODE REVIEW

File: code/modules/power/apc/apc_tool_act.dm [hunk 1]
```diff
@@ -1,4 +1,221 @@
...
+	if(has_electronics)
+		balloon_alert(user, "there is already a board!")
+		return ITEM_INTERACT_BLOCKING
+
+	if(machine_stat & BROKEN)
+		balloon_alert(user, "the frame is damaged!")
+		return ITEM_INTERACT_BLOCKING
+
+	user.visible_message(span_notice("[user.name] inserts the power control board into [src]."))
+	balloon_alert(user, "you start to insert the board...")
```
Comment (san7890) [inline code/modules/power/apc/apc_tool_act.dm:140]:
```suggestion
	balloon_alert(user, "inserting the board...")
```

more succinct imo

File: code/modules/power/apc/apc_tool_act.dm [hunk 2]
```diff
@@ -1,4 +1,221 @@
...
+/// Called when we interact with the APC with an electroadaptive pseudocircuit, used by cyborgs to install a board or weak cell
+/obj/machinery/power/apc/proc/pseudocircuit_act(mob/living/user, obj/item/electroadaptive_pseudocircuit/pseudocircuit)
+	if(!has_electronics)
+		if(machine_stat & BROKEN)
+			balloon_alert(user, "frame is too damaged!")
+			return ITEM_INTERACT_BLOCKING
+		if(!pseudocircuit.adapt_circuit(user, circuit_cost = 50 KILO JOULES))
+			return ITEM_INTERACT_BLOCKING
+		user.visible_message(span_notice("[user] fabricates a circuit and places it into [src]."), \
+		span_notice("You adapt a power control board and click it into place in [src]'s guts."))
```
Comment (san7890) [inline code/modules/power/apc/apc_tool_act.dm:161]:
```suggestion
		user.visible_message(
			span_notice("[user] fabricates a circuit and places it into [src]."),
			span_notice("You adapt a power control board and click it into place in [src]'s guts."),
		)
```

let's multiline it right

File: code/modules/power/apc/apc_tool_act.dm [hunk 3]
```diff
@@ -1,4 +1,221 @@
...
+		if(machine_stat & MAINT)
+			balloon_alert(user, "no board for a cell!")
+			return ITEM_INTERACT_BLOCKING
+		if(!pseudocircuit.adapt_circuit(user, circuit_cost = 500 KILO JOULES))
+			return ITEM_INTERACT_BLOCKING
+		var/obj/item/stock_parts/cell/crap/empty/bad_cell = new(src)
+		bad_cell.forceMove(src)
+		cell = bad_cell
+		user.visible_message(span_notice("[user] fabricates a weak power cell and places it into [src]."), \
+		span_warning("Your [pseudocircuit.name] whirrs with strain as you create a weak power cell and place it into [src]!"))
```
Comment (san7890) [inline code/modules/power/apc/apc_tool_act.dm:176]:
```suggestion
		user.visible_message(
			span_notice("[user] fabricates a weak power cell and places it into [src]."),
			span_warning("Your [pseudocircuit.name] whirrs with strain as you create a weak power cell and place it into [src]!"),
		)
```

File: code/modules/power/apc/apc_tool_act.dm [hunk 4]
```diff
@@ -1,4 +1,221 @@
...
+		user.visible_message(span_notice("[user.name] replaces missing APC's cover."))
+		balloon_alert(user, "replacing APC's cover...")
+		if(do_after(user, 20, target = src)) // replacing cover is quicker than replacing whole frame
+			balloon_alert(user, "cover replaced")
+			qdel(wallframe)
+			update_integrity(30) //needs to be welded to fully repair but can work without
+			set_machine_stat(machine_stat & ~(BROKEN|MAINT))
+			opened = APC_COVER_OPENED
+			update_appearance()
+			return ITEM_INTERACT_SUCCESS
```
Comment (san7890) [inline code/modules/power/apc/apc_tool_act.dm:201]: ugly indentation here, invert it to an early return `if(!do_after(...))`

File: code/modules/power/apc/apc_tool_act.dm [hunk 5]
```diff
@@ -1,4 +1,221 @@
...
+		return ITEM_INTERACT_BLOCKING
+	if(has_electronics)
+		balloon_alert(user, "remove the board inside!")
+		return ITEM_INTERACT_BLOCKING
+	user.visible_message(span_notice("[user.name] replaces the damaged APC frame with a new one."))
+	balloon_alert(user, "replacing damaged frame...")
+	if(do_after(user, 50, target = src))
```
Comment (san7890) [inline code/modules/power/apc/apc_tool_act.dm:208]: ditto early return

**Review by san7890 — CHANGES_REQUESTED** [review]
this stuff probably isn't yours but it's appreciated to touch it up in this refactor

**Comment by 00-Steven** [comment]
Agh, yeah those multilines especially are ugly as shit.
Originally chose not to affect the code within the sections cause it's basically entirely green lines and I felt like that'd suck ass to review in the same pr, but yeah fair enough.

**Comment by 00-Steven** [comment]
cool cool

**Pull request merged**